### PR TITLE
Fix whitespace strings returning true (#3)

### LINKED ISSUE #3: Whitespace strings return true
Unfortunately the empty string is not the only one that casts to zero without containing a number:
```javascript
isNumber('   ') // true
isNumber('\r\n\t') // true
```

I guess there need to be some whitespace strings in the test arrays. Probably also some strings with numbers AND whitespace, which are (correctly) recognized as numbers currently:
```javascript
isNumber('   56\r\n  ') // true
```

FWIW, for my own purposes I decided I was OK abandoning the Number and String objects - `n = new Number(1)` and `n = new String('1')` - and so was able to simplify to:

```javascript
module.exports = function(n) {
    if(n === '') return false;

    var type = typeof n;
    if(type === 'string') n = +n;
    else if(type !== 'number') return false;

    return n - n < 1;
};
```
which runs 2-3x faster than the more general case using `kind-of`. Notice in particular that you only have to cast if you don't already have a number.

And then I could handle the whitespace without too much overhead by:

```javascript
module.exports = function(n) {
    var type = typeof n;
    if(type === 'string') {
        var original = n;
        n = +n;
        // whitespace strings cast to zero - filter them out
        if(n===0 && !original.trim()) return false;
    }
    else if(type !== 'number') return false;

    return n - n < 1;
}
```

### PR DESCRIPTION
**Pull request by tunnckoCore**
ref #3 

- micro optimization for falsey values
- update benchmark, make it work (even w/o updating the version it was broken in current master, idk)
- fixes #3, minimal perf regression for string benchmarks, because the `.trim()` i suppose. but i think there's no other way to fix #3 without a bit of regression

### CODE REVIEW

**Comment by tunnckoCore** [comment]
And btw, got to same benchmark results without deps. And it's a bit better because in current there's is a bit require time and it hits few `if`s in kind-of.

File: index.js [hunk 1]
```diff
@@ -10,10 +10,17 @@
 var typeOf = require('kind-of');
 
 module.exports = function isNumber(num) {
+  if (!num && num !== 0) return false;
```
Comment (jonschlinkert) [inline index.js]: Is this line necessary? (I'm really asking, I might be missing something) seems like we would catch that with existing conditions

Comment (tunnckoCore) [inline index.js]: Return earlier concept. Not sure enough but yea as i said, micro optimization - it just won't hit the kind-of if it is null or undef. It makes little sense to have it.

File: index.js [hunk 2]
```diff
@@ -10,10 +10,17 @@
...
   var type = typeOf(num);
   if (type !== 'number' && type !== 'string') {
     return false;
   }
+  if (type === 'string') {
+    num = num.trim();
+    if (!num.length) return false;
+  }
+
   var n = +num;
```
Comment (tunnckoCore) [inline index.js]: Hm. It's even better if we remove 19-23 and add

```js
if (n === 0 && type === 'string' && !num.trim()) return false;
```
below `var n = +num`

Not sure enough why but... tests are still passing.

Comment (jonschlinkert) [inline index.js]: `if (n === 0 && type === 'string' && !num.trim()) return false;` this can never be true. Since `0` will never be a string.

**Comment by jonschlinkert** [comment]
```js
var typeOf = require('kind-of');

module.exports = function isNumber(num) {
  var type = typeOf(num);

  if (type === 'string') {
    if (!num.trim()) return false;
  } else if (type !== 'number') {
    return false;
  }

  return (num - num + 1) >= 0;
};
```

Benchmarks

```sh
# benchmark/fixtures/all.js (1105 bytes)
  current x 106,145 ops/sec ±0.79% (87 runs sampled)

# benchmark/fixtures/infinity.js (31 bytes)
  current x 16,485,894 ops/sec ±1.25% (86 runs sampled)

# benchmark/fixtures/null.js (27 bytes)
  current x 19,949,726 ops/sec ±1.58% (82 runs sampled)

# benchmark/fixtures/number.js (24 bytes)
  current x 18,371,482 ops/sec ±1.03% (89 runs sampled)

# benchmark/fixtures/string-number.js (26 bytes)
  current x 7,291,852 ops/sec ±0.87% (88 runs sampled)

# benchmark/fixtures/string.js (28 bytes)
  current x 6,463,536 ops/sec ±0.97% (85 runs sampled)
```

**Comment by tunnckoCore** [comment]
Actually, yea haha. Forgot that it's the crazier lang ever and it can `-` two strings :D 

I'll update.

**Pull request merged**